packages/xo-server/src/xapi/index.mjs

```diff
@@ -958,14 +958,21 @@ export default class Xapi extends XapiBase {
     }
   }
 
-  async startVm(vmId, options) {
+  /**
+   *
+   * @param {string} vmId
+   * @param {object} options
+   * @param {boolean} [options.startOnly] - If true, don't try to unpause/resume the VM if VM_BAD_POWER_STATE is thrown
+   *
+   */
+  async startVm(vmId, { startOnly = false, ...options } = {}) {
     try {
       await this._startVm(this.getObject(vmId), options)
     } catch (e) {
       if (e.code === 'OPERATION_BLOCKED') {
         throw forbiddenOperation('Start', e.params[1])
       }
-      if (e.code === 'VM_BAD_POWER_STATE') {
+      if (e.code === 'VM_BAD_POWER_STATE' && !startOnly) {
         return e.params[2] === 'paused' ? this.unpauseVm(vmId) : this.resumeVm(vmId)
```

Comment:
can the vm be in a running state  here ?

Reply:
Yes

Reply:
I think this is a bug, but the error message is not completely wrong. I let you choose if you prefer to fix it, or just add a comment